proposals/code-signing-for-homebrew.md

```diff
@@ -0,0 +1,101 @@
+3. The signing step is responsible for uploading the resulting signature to the same object storage as the bottle, and ensuring that the uploaded signature is addressable in a consistent and publicly derivable manner from just the formula and bottle's own metadata. In other words: a client that knows how to access the bottle's resource should also be able to access the bottle's signature resource with no additional metadata. 
+
+At the end of this flow, both the bottle and its signature should appear in the tap's associated object storage. 
+
+This flow does not sign for bottles that have already been produced, meaning that pre-existing formulae will not receive bottle signatures until they are rebuilt or otherwise updated. Consequently, each tap needs to establish a piece of "flag day" metadata: a tap should designate a timestamp `T`, after which all bottles defined by formulae in the tap are signed. This fundamentally requires that the tap itself is honest about its flag day: an attacker who can replace the tap's flag day can perform a downgrade by disabling signature verification. This is no different in scope from Homebrew's existing threat model, where an attacker must not be allowed to modify files in a tap. In other words: this signing scheme only protects the bottles themselves, not the tap's repository; the tap's repository must continue to be protected as a matter of policy.
+
+### Sigstore signatures for third-party taps
+
+While not in scope for this proposal, this work would also enable signing for third-party taps, such as those maintained by independent open source organizations or companies.
+
+For third-party taps hosted on GitHub, the signing flow would be identical: any bottle-building workflow (including reusing the official one) would sign for bottles.
+
+## Client-side signature verification in Homebrew
+
+The Sigstore signatures described above are submitted to each tap's object storage so that they can be verified by clients during the normal `brew install` flow.
+
+To verify a Sigstore signature, a particular `brew` client needs the following:
+
+- A trust proposition, i.e. a reason to trust the identity producing the signature. For Homebrew, this trust proposition comes from the tap's own identity: the act of performing `brew tap foo/bar` establishes that Sigstore identities tied to `github.com/foo/bar` can be trusted to sign for bottles in that tap. This applies to implicit `brew tap` invocations as well, e.g. `brew install foo/bar/baz` asserts that the user expects a signature on `baz` from the `foo/bar` GitHub identity.
```

Comment:
This is great! Love that identity can be automatically derived.

To confirm, there would be no support for user identities, only GitHub Actions identities?

Reply:
Yep, that was my thought for the MVP: the Ruby DSL specified below in the proposal would give us the flexibility to support non-GHA identities in the future, but the 99.9% case (IMO) doesn't need them.

Reply:
Yea, it'd be worth a brief mention that the focus is on CI identities.

Reply:
https://github.com/ossf/wg-securing-software-repos/pull/21/commits/0f9675a427973d84ee56f84c821b12dd526c987c